prusti-interface/src/specs/checker/predicate_checks.rs

```diff
@@ -28,17 +28,6 @@ impl<'tcx> SpecCheckerStrategy<'tcx> for IllegalPredicateUsagesChecker {
             self.collect_illegal_predicate_usages(collected_predicates.predicates, env.query);
         debug!("Predicate usages: {:?}", illegal_pred_usages);
 
-        let body_errors = collected_predicates
-            .abstract_predicate_with_bodies
-            .into_iter()
-            .map(|def_id| {
-                let span = env.query.get_def_span(def_id);
-                PrustiError::incorrect(
-                    "abstract predicates must not have bodies".to_string(),
-                    MultiSpan::from_span(span),
-                )
-            });
```

Comment:
Ah, we forbid this originally because behavioural subtyping with predicates is not checked. Maybe we should put a TODO here at least.

Reply:
Are you saying abstract predicate implementations should ideally be checked for behavioral subtyping against the default implementation (if present)? I'm not sure how that would work, since the implication would be flipped depending on whether the predicate is used in a pre- or postcondition, for example, right?